.github/workflows/release.yml

```diff
@@ -1,38 +1,37 @@
-# .github/workflows/release.yml
-name: goreleaser
+name: Release
 
 on:
   push:
     # run only against tags
     tags:
-      - "*"
+      - "v*"
 
 permissions:
   contents: write
-  # packages: write
-  # issues: write
 
 jobs:
   goreleaser:
+    name: Release with GoReleaser
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout code
+        uses: actions/checkout@v4
         with:
           fetch-depth: 0
-      - run: git fetch --force --tags
-      - uses: actions/setup-go@v4
+      - name: Fetch tags
+        run: git fetch --force --tags
+      - name: Set up Go
+        uses: actions/setup-go@v5
         with:
-          go-version: stable
-      # More assembly might be required: Docker logins, GPG, etc.
-      # It all depends on your needs.
-      - uses: goreleaser/goreleaser-action@v5
+          go-version: "1.24"
+          cache: true
+      - name: Run tests before release
+        run: go test ./...
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v6
         with:
-          # either 'goreleaser' (default) or 'goreleaser-pro':
           distribution: goreleaser
           version: latest
           args: release --clean
         env:
           GITHUB_TOKEN: ${{ secrets.STOIC_SECRET }}
-          # Your GoReleaser Pro key, if you are using the 'goreleaser-pro'
-          # distribution:
-          # GORELEASER_KEY: ${{ secrets.GORELEASER_KEY }}
```

Comment:
Please don't remove this. I have  goreleaser PRO license, I was planning to start using that at some point.

Reply:
Can do. It is the habit of a line removal focus and a YAGNI leftover.